### ISSUE TITLE
Error in python: free(): invalid pointer

### ISSUE THREAD

**Issue by Enamex** [pytorch/pytorch#5470]
OS: ubuntu 14
- PyTorch version: 0.3.0
- How you installed PyTorch: conda
- Python version: 3.6.4
- CUDA/cuDNN version:  9.0.176
- GPU models: Titan X

the following code crashes after  self.lstm1(embed_atten,  (output1, state1))

```python
def forward(self, video_feat, video_mask, captions, caption_mask):
        state1  = Variable(torch.FloatTensor(self.batch_size, self.dim_hidden).zero_().cuda(), requires_grad=False)
        state2  = Variable(torch.FloatTensor(self.batch_size, self.dim_hidden).zero_().cuda(), requires_grad=False)

        output1 = Variable(torch.FloatTensor(self.batch_size, self.dim_hidden).zero_().cuda(), requires_grad=False)
        output2 = Variable(torch.FloatTensor(self.batch_size, self.dim_hidden).zero_().cuda(), requires_grad=False)
        vid_emb = video_feat.permute(1, 0, 2) # n x b x h
        h_prev  = vid_emb[0]
        total_loss = 0.

        # n x h x 1
        brcst_w = self.embed_att_w.unsqueeze(0).expand(self.n_lstm_steps, self.embed_att_w.size(0), self.embed_att_w.size(1))
        # n x h x h
        att_Ua  =  self.embed_att_Ua.unsqueeze(0).expand(self.n_lstm_steps, self.embed_att_Ua.size(0), self.embed_att_Ua.size(1))

        #print("UA: {} | Ba: {} | Emb: {}".format(att_Ua.size(), self.embed_att_ba.size(), vid_emb.size()))
        # n x b x h
        image_part = torch.bmm(vid_emb, att_Ua) + self.embed_att_ba

        for i in range(self.n_lstm_steps):
            ##### soft temporal attention #####
            # n x b x h
            e_i = torch.tanh(torch.mm(h_prev, self.embed_att_Wa) + image_part)
            # n x b x 1
            e_i = torch.bmm(e_i, brcst_w)
            # n x b
            e_i = torch.sum(e_i, 2)
            # n x b
            e_hat_exp = torch.mul(torch.t(video_mask), torch.exp(e_i))
            # b
            denom = torch.sum(e_hat_exp, 0)
            denom = denom + torch.eq(denom, 0.0).cuda().float()
            # n x b x h
            e_div = torch.div(e_hat_exp, denom)
            alphas = e_div.unsqueeze(2).expand(e_div.size(0), e_div.size(1), self.dim_vid)
            # n x b x h
            attention_list = torch.mul(alphas, vid_emb)
            # b x h  
            atten = torch.sum(attention_list,0)

            ##### lstm #####
            current_embed   = self.Wemb(captions[:, i])

            embed_atten = torch.cat((current_embed, atten), 1)
            output1, state1 = self.lstm1(embed_atten ,  (output1, state1))
           
            output2, state2 = self.lstm2(output1, (output2, state2))
            h_prev = output2
           
            logit_words = self.embed_word(output2)
            loss = nn.CrossEntropyLoss()

            cross_entropy = loss(logit_words, captions[:, i+1])
            cross_entropy = cross_entropy * caption_mask[:, i+1]

            total_loss = total_loss + (torch.sum(cross_entropy) / self.batch_size)

        return total_loss
```

**Comment by zou3519** [comment]
Could you please provide sample inputs for `video_feat, video_mask, captions, caption_mask` (or just their sizes?)

**Comment by karimhasebou** [comment]
@zou3519 Video_feat is 4x80x500, video_mask is 4x80, captions 4x1x80, caption mask 4x1x80

**Comment by soumith** [comment]
@Enamex @karimhasebou this is an error message issue that we have in binary wheel builds. Can you install pytorch from `conda` and rerun your script (or alternatively, can you build [pytorch from source](https://github.com/pytorch/pytorch#from-source) and rerun your script)

**Comment by karimhasebou** [comment]
@soumith  i used conda to install pytorch. I have tried upgrading anaconda and reinstalling pytorch but that didn't fix it

**Comment by zou3519** [comment]
@karimhasebou could you try building from source? You can check out the v0.3.1 branch if master is too much of a change at the moment. I believe soumith is referring to [this issue](https://github.com/pytorch/pytorch/issues/5400) which I think happens on both conda installs and the whls.

**Comment by BKHMSI** [comment]
@karimhasebou and I figured out the problem, the size mismatch between `embed_atten` and `output1, state1` in the LSTMCell caused this behavior.

**Comment by soumith** [comment]
I'm using https://github.com/pytorch/pytorch/issues/5400 as the master issue to track the Invalid Free on error message, closing this issue in favor of #5400
